### PR TITLE
iox-#1533 Standalone roudi environment

### LINKED ISSUE #1533: Roudi Environment independent from Googletest
## Brief feature description

Currently, the `iceoryx_posh_testing` library depends on `iceoryx_hoofs_testing` which depends itself on the GoogleTest library.
The RouDi environment is part of the posh testing library and offers a GTest "Binding" to make use of RouDi in GoogleTests: https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_posh/testing/include/iceoryx_posh/testing/roudi_gtest.hpp

## Detailed information

The RouDi environment should maybe put into a separate library that is independent from GoogleTest to make it usable by other testing Frameworks.

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

This PR creates a separate package for the `RouDiEnvironment` which does not depend on `gTest` and can also be build without the tests. This is important for the Rust bindings since they need the RouDi environment for its own tests but not the iceoryx C++ tests which take quite long to build. 

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #1533

### CODE REVIEW

File: iceoryx_posh/roudi_env/include/iceoryx_posh/roudi_env/minimal_roudi_config.hpp [hunk 1]
```diff
@@ -14,15 +14,15 @@
...
+#ifndef IOX_POSH_ROUDI_ENV_MINIMAL_ROUDI_CONFIG_HPP
+#define IOX_POSH_ROUDI_ENV_MINIMAL_ROUDI_CONFIG_HPP
 
 #include "iceoryx_posh/iceoryx_posh_config.hpp"
 #include "iox/builder.hpp"
 
 namespace iox
 {
-namespace testing
+namespace roudi_env
```
Comment (elBoberido) [inline iceoryx_posh/roudi_env/include/iceoryx_posh/roudi_env/minimal_roudi_config.hpp:25]: I would also be fine to keep the `testing` namespace but it felt somehow better to move the `RouDiEnvironment` to its own namespace

File: iceoryx_posh/roudi_env/include/iceoryx_posh/roudi_env/roudi_env.hpp
Comment (elBoberido) [inline iceoryx_posh/roudi_env/include/iceoryx_posh/roudi_env/roudi_env.hpp:1]: This file is actually moved and the history preserved but github does not manage to detect it

File: iceoryx_posh/roudi_env/source/roudi_env.cpp
Comment (elBoberido) [inline iceoryx_posh/roudi_env/source/roudi_env.cpp:1]: Same here, the file is moved and the history is preserved

File: iceoryx_posh/roudi_env/include/iceoryx_posh/roudi_env/roudi_env.hpp
Comment (elBoberido) [inline iceoryx_posh/roudi_env/include/iceoryx_posh/roudi_env/roudi_env.hpp:1]: This file is actually moved and the history preserved but github does not manage to detect it

File: iceoryx_posh/roudi_env/source/roudi_env.cpp
Comment (elBoberido) [inline iceoryx_posh/roudi_env/source/roudi_env.cpp:1]: Same here, the file is moved and the history is preserved

File: doc/website/release-notes/iceoryx-unreleased.md [hunk 2]
```diff
@@ -1181,3 +1182,33 @@
...
+
+    // before
+    roudiEnv.InterOpWait();
+
+    // after
+    roudiEnv.triggerDiscoveryLoopAndWaitToFinish();
+
+
+    // before
+    SetInterOpWaitingTime(/*chrono*/);
```
Comment (FerdinandSpitzschnueffler) [inline doc/website/release-notes/iceoryx-unreleased.md:1207]: Why not `roudiEnv.SetInterOpWaitingTime(/*chrono*/)`? I haven't looked it up yet, but I'd expect this is also a method of the roudi environment.

**Review by FerdinandSpitzschnueffler — COMMENTED** [review]
Looks good to me, but maybe @dkroenke could have a look at the cmake and bazel files.

File: doc/website/release-notes/iceoryx-unreleased.md (hunk 2 above)
Comment (elBoberido) [reply doc/website/release-notes/iceoryx-unreleased.md:1207]: No, I have deprecated `InterOpWait`/`SetInterOpWaitingTime` and replaced them with `triggerDiscoveryLoopAndWaitToFinish`/`setDiscoveryLoopWaitToFinishTimeout`. The former did not fully state when it is needed and I found some cases where `InterOpWait` was added in tests without being required. I hope the more expressive names make it more clear when and especially when not to use

**Comment by elBoberido** [comment]
@mossmaurice @dkroenke @MatthiasKillat @elfenpiff 

I came to the conclusion that it was not one of my brightest moments to add all the committer as reviewer since everybody would think that another one would already review. My new approach is now to ping all the committer in a comment and the ones who like to do the review can add themself as reviewer. I hope this makes it more clear when a PR has sufficient reviewers :)

File: doc/website/release-notes/iceoryx-unreleased.md (hunk 2 above)
Comment (FerdinandSpitzschnueffler) [reply doc/website/release-notes/iceoryx-unreleased.md:1207]: Maybe a misunderstanding. I meant that I'd expected the (old and new) method to be called on the `roudiEnv` instance, like `roudiEnv.method()`.

Comment (elBoberido) [reply doc/website/release-notes/iceoryx-unreleased.md:1207]: Oh, sure :sweat_smile: 

Depending on what one uses it is either `roudiEnv.method()` or  when `RouDi_GTest` is used just `method`.

Comment (elBoberido) [reply doc/website/release-notes/iceoryx-unreleased.md:1207]: ... and you are right that for consistency reasons it should mirror `triggerDiscoveryLoopAndWaitToFinish`

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

**Review by MatthiasKillat — APPROVED** [review]

**Pull request merged**